As far as I understand the error output is slightly different on darwin which causes the test to fail:

```
self = <test.test_error.TestErrorConversion object at 0x110663910>

    def test_error_message_reuse(self):
        """
        Test if the error message string is retained if the caught exception
        has more than one element in `args`.
        """
        # See ticket #76.
        with pytest.raises(ftputil.error.FTPOSError) as exc_info:
            # Format "host:port" doesn't work. The use here is intentional.
            host = ftputil.FTPHost("localhost:21", "", "")
        exc = exc_info.value
        # The error message may be different for different Python versions.
>       assert "No address associated with hostname" in str(
            exc
        ) or "Name or service not known" in str(exc)
E       AssertionError: assert ('No address associated with hostname' in '[Errno 8] nodename nor servname provided, or not known\nDebugging info: ftputil 4.0.0, Python 3.8.6 (darwin)' or 'Name or service not known' in '[Errno 8] nodename nor servname provided, or not known\nDebugging info: ftputil 4.0.0, Python 3.8.6 (darwin)')
E        +  where '[Errno 8] nodename nor servname provided, or not known\nDebugging info: ftputil 4.0.0, Python 3.8.6 (darwin)' = str(FTPOSError(8, 'nodename nor servname provided, or not known'))
E        +  and   '[Errno 8] nodename nor servname provided, or not known\nDebugging info: ftputil 4.0.0, Python 3.8.6 (darwin)' = str(FTPOSError(8, 'nodename nor servname provided, or not known'))

test/test_error.py:66: AssertionError
```

--- a/pkgs/development/python-modules/ftputil/default.nix
+++ b/pkgs/development/python-modules/ftputil/default.nix
@@ -1,4 +1,4 @@
-{ lib, buildPythonPackage, fetchPypi, pythonOlder, pytest, freezegun }:
+{ stdenv, lib, buildPythonPackage, fetchPypi, pythonOlder, pytest, freezegun }:
 
 buildPythonPackage rec {
   version = "4.0.0";
@@ -18,12 +18,15 @@ buildPythonPackage rec {
     py.test test \
       -k "not test_public_servers and not test_real_ftp \
           and not test_set_parser and not test_repr \
-          and not test_conditional_upload and not test_conditional_download_with_older_target"
-  '';
+          and not test_conditional_upload and not test_conditional_download_with_older_target \
+  ''
+  # need until https://ftputil.sschwarzer.net/trac/ticket/140#ticket is fixed
+  + lib.optionalString stdenv.isDarwin ''and not test_error_message_reuse''
+  + ''"'';
 
   meta = with lib; {
     description = "High-level FTP client library (virtual file system and more)";
-    homepage    = "http://ftputil.sschwarzer.net/";
-    license     = licenses.bsd2; # "Modified BSD license, says pypi"
+    homepage = "http://ftputil.sschwarzer.net/";
+    license = licenses.bsd2; # "Modified BSD license, says pypi"
   };
 }

